app/controllers/posts_controller.rb

```diff
@@ -24,11 +24,28 @@ def show
         @post = Post.find(params[:id])
     end
 
+    def edit 
+        @post = Post.find(params[:id])
+        unless current_user == @post.user
+            redirect_to @post, notice: "You cannot edit this post"
```

Comment:
Aim for helpful error error messages, considering who they are for. If the message is in your logs, then make it helpful to you as a developer. If it's displayed to the user, make it helpful to them. As a user, I wouldn't know whether this was because something was wrong with the site, vs something I did. A message user should explain either what went wrong and what can be done to fix the situation, or it should explain why repeating the action won't ever work. In this case, something like "Sorry, you can't edit this post because it isn't your post"